Source: http://git.postgresql.org/gitweb/?p=postgresql.git;a=commitdiff;h=ccd69b88
Upstream: Yes
Reason: Fix building postgresql[perl] with perl 5.14
---
 src/pl/plperl/plperl.c |    2 +-
 src/pl/plperl/plperl.h |    5 +++++
 2 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/src/pl/plperl/plperl.c b/src/pl/plperl/plperl.c
index f2e8ad2..d2c672c 100644
--- a/src/pl/plperl/plperl.c
+++ b/src/pl/plperl/plperl.c
@@ -926,7 +926,7 @@ plperl_trusted_init(void)
 		if (!isGV_with_GP(sv) || !GvCV(sv))
 			continue;
 		SvREFCNT_dec(GvCV(sv)); /* free the CV */
-		GvCV(sv) = NULL;		/* prevent call via GV */
+		GvCV_set(sv, NULL);		/* prevent call via GV */
 	}
 	hv_clear(stash);
 
diff --git a/src/pl/plperl/plperl.h b/src/pl/plperl/plperl.h
index a375bb5..c1236b7 100644
--- a/src/pl/plperl/plperl.h
+++ b/src/pl/plperl/plperl.h
@@ -85,6 +85,11 @@
 								(U32)HeKUTF8(he))
 #endif
 
+/* supply GvCV_set if it's missing - ppport.h doesn't supply it, unfortunately */
+#ifndef GvCV_set
+#define GvCV_set(gv, cv)		(GvCV(gv) = cv)
+#endif
+
 /* declare routines from plperl.c for access by .xs files */
 HV		   *plperl_spi_exec(char *, int);
 void		plperl_return_next(SV *);
-- 
1.7.2.5

